src/utils/sshUtils.ts

```diff
@@ -64,3 +89,15 @@ export async function configureSshConfig(vmti: VirtualMachineTreeItem, sshKeyPat
     await fse.writeFile(sshConfigPath, configFile);
     ext.outputChannel.appendLog(localize('addingEntry', `Added new entry to "{0}" with Host "{1}".`, sshConfigPath, host));
 }
+
+async function sshKeygenExists(): Promise<boolean> {
+    try {
+        // this command should _always_ fail.  If ssh-keygen exists, it will throw an unknown option error.  If it doesn't, then it'll throw the not recognized error.
```

Comment:
How about we use this package so we don't have to rely on an exact error message:

https://www.npmjs.com/package/which